lib/api/api.js

```diff
@@ -67,6 +72,17 @@ const api = {
             // Pull the source bucket and source object separated by /
             sourceBucket = source.slice(0, slashSeparator);
             sourceObject = source.slice(slashSeparator + 1);
+            sourceVersionId = query ?
+                querystring.parse(query).versionId : undefined;
+            sourceVersionId = sourceVersionId || undefined;
+            if (sourceVersionId && sourceVersionId !== 'null') {
+                try {
+                    sourceVersionId = VID.decrypt(sourceVersionId);
+                } catch (exception) {
+                    return callback(errors.InvalidArgument
+                        .customizeDescription('Invalid version id specified'));
+                }
+            }
```

Comment:
It looks like we'll have to update Arsenal to update the request contexts.
But I also see we already pass the request headers, which contain the version id, to Arsenal. Do we still want to add another parameter?
Do we currently use these condition keys, or we're just adding them to request contexts / Arsenal for future use?